src/Random/InverseTransformSampling.h

```diff
@@ -0,0 +1,173 @@
+// -*- C++ -*-
+/***************************************************************************
+ *
+ * The IPPL Framework
+ *
+ * This program was prepared by PSI.
+ * All rights in the program are reserved by PSI.
+ * Neither PSI nor the author(s)
+ * makes any warranty, express or implied, or assumes any liability or
+ * responsibility for the use of this software
+ *
+ * Visit www.amas.web.psi for more details
+ *
+ ***************************************************************************/
+
+// Class InverseTransformSampling
+//   This class can be used for generating samples of a given distribution function class using
+//   inverse of its cumulative distribution on host or device.
+//
+
+#ifndef IPPL_INVERSE_TRANSFORM_SAMPLING_H
+#define IPPL_INVERSE_TRANSFORM_SAMPLING_H
+
+#include "Random/Utility.h"
```

Comment:
I am not sure why this should be confusing. Please let me know which of the solutions you prefer and I'll do it.

Reply:
I am not sure @matt-frey what do you think? Do we need any changes like I mentioned before or it may be fine as it is?

Reply:
As this file only contains the Newton-Raphson method, I would rename it appropriately.

Reply:
I kept the name general (utility) so that in case in the future other useful mathematical functions/structs for sampling can be added there. If that's not desirable, let me know so I change the name to NewtonRaphson.h or something like that